packages/@aws-cdk/aws-batch/lib/job-definition.ts

```diff
@@ -0,0 +1,245 @@
+   */
+  fromNodeIndex?: number;
+
+  /**
+   * The maximum node index value to apply this container definition against. If omitted, the highest value is used relative.
+   *
+   * to the number of nodes associated with the job. You may nest node ranges, for example 0:10 and 4:5,
+   * in which case the 4:5 range properties override the 0:10 properties.
+   *
+   * @default {@link IMultiNodeprops.count}
+   */
+  toNodeIndex?: number;
+}
+
+/**
+ * An interface representing a job definition - either a new one, created with the CDK, *using the
+ * {@link JobDefinition} class, or existing ones, referenced using the {@link JobDefinition.fromJobDefinitionArn} method..
+ */
+export interface IJobDefinition extends IResource {
+  /**
+   * The ARN of this batch job definition
+   *
+   * @attribute
+   */
+  readonly jobDefinitionArn: string;
+
+  /**
+   * The name of the batch job definition
+   *
+   * @attribute
+   */
+  readonly jobDefinitionName: string;
+}
+
+/**
+ * Batch Job Definition
+ *
+ * Defines a batch job definition to execute a specific batch job.
+ */
+export class JobDefinition extends Resource implements IJobDefinition {
+  /**
+   * Imports an existing batch job definition by its amazon resource name.
+   *
+   * @param scope
+   * @param id
+   * @param jobDefinitionArn
+   */
+  public static fromJobDefinitionArn(scope: Construct, id: string, jobDefinitionArn: string): IJobDefinition {
+    const stack = Stack.of(scope);
+    const jobDefName = stack.parseArn(jobDefinitionArn).resource;
+
+    class Import extends Resource implements IJobDefinition {
+      public readonly jobDefinitionArn = jobDefinitionArn;
+      public readonly jobDefinitionName = jobDefName;
+    }
+
+    return new Import(scope, id);
+  }
+
+  public readonly jobDefinitionArn: string;
+  public readonly jobDefinitionName: string;
+  private readonly imageConfig: IJobDefinitionImageConfig;
+
+  constructor(scope: Construct, id: string, props: JobDefinitionProps) {
+    super(scope, id, {
+      physicalName: props.jobDefinitionName,
+    });
+
+    this.imageConfig = new JobDefinitionImageConfig(this, props.container);
+
+    const jobDef = new CfnJobDefinition(this, 'Resource', {
+      jobDefinitionName: props.jobDefinitionName,
+      containerProperties: this.buildJobContainer(props.container),
+      type: 'container',
+      nodeProperties: props.nodeProps
+      ? { mainNode: props.nodeProps.mainNode,
+          nodeRangeProperties: this.buildNodeRangeProps(props.nodeProps),
+          numNodes: props.nodeProps.count }
+      : undefined,
+      parameters: props.parameters,
+      retryStrategy: {
+        attempts: props.retryAttempts || 1,
+      },
+      timeout: {
+        attemptDurationSeconds: props.timeout ? props.timeout.toSeconds() : undefined,
+      },
+    });
+
+    this.jobDefinitionArn = this.getResourceArnAttribute(jobDef.ref, {
+      service: 'batch',
+      resource: 'job-definition',
+      resourceName: this.physicalName,
```

Comment:
Again, there's a discrepancy between the `from` method, and the ARN constructed here.

Reply:
I am unsure how I should change this or so. Could you provide an example?

Reply:
This is the same as my comment here: https://github.com/aws/aws-cdk/pull/4732#discussion_r363457612